analysis/quantifier-comparisons/01-import-quant-data.Rmd

```diff
@@ -147,6 +232,235 @@ alevin_sces <- alevin_quant_ids  %>%
+  dplyr::pull(quant_dir)
+
+alevin_fry_sces <- alevin_fry_quant_ids  %>%
+  purrr::map(
+    ~ tximport(file.path(alevin_fry_data_dir, .x, "alevin", "quants_mat.gz"),
+               type = "alevin")) %>%
+  purrr::map(
+    ~ SingleCellExperiment(list(counts = .x$counts))) 
+
+# add names
+names(alevin_fry_sces) <- alevin_fry_quant_ids
+```
+
+```{r}
+# collapse intron counts for only counts matrices containing intron regions
+alevin_fry_nucleus_sces <- alevin_fry_sces[grep("spliced_intron", names(alevin_fry_sces))] %>%
+  purrr::map(
+    ~ counts(.x)
+  ) %>%
+  purrr::map(
+    ~ collapse_intron_counts(.x)
+  ) %>%
+  purrr::map(
+    ~ SingleCellExperiment(list(counts = .x))
+  )
+
+# replace the sces in the original list with the collapsed sces
+alevin_fry_sces[which(names(alevin_fry_sces) %in% names(alevin_fry_nucleus_sces))] <- 
+  alevin_fry_nucleus_sces
+```
+
+
+```{r}
+## alevin fry, no permit list, with sketch 
+alevin_fry_sketch_quant_ids <- quant_info %>%
+  dplyr::filter(tool == "alevin_fry" & alevin_permit_list != "unfiltered" & alevin_alignment == "sketch") %>%
+  dplyr::pull(quant_dir)
+
+alevin_fry_sketch_sces <- alevin_fry_sketch_quant_ids  %>%
+  purrr::map(
+    ~ tximport(file.path(alevin_fry_data_dir, .x, "alevin", "quants_mat.gz"),
+               type = "alevin")) %>%
+  purrr::map(
+    ~ SingleCellExperiment(list(counts = .x$counts))) 
+
+# add names
+names(alevin_fry_sketch_sces) <- alevin_fry_sketch_quant_ids
+```
+
+```{r}
+# collapse intron counts for only counts matrices containing intron regions
+alevin_fry_sketch_nucleus_sces <- alevin_fry_sketch_sces[grep("spliced_intron", names(alevin_fry_sketch_sces))] %>%
+  purrr::map(
+    ~ counts(.x)
+  ) %>%
+  purrr::map(
+    ~ collapse_intron_counts(.x)
+  ) %>%
+  purrr::map(
+    ~ SingleCellExperiment(list(counts = .x))
+  )
+
+# replace the sces in the original list with the collapsed sces
+alevin_fry_sketch_sces[which(names(alevin_fry_sketch_sces) %in% names(alevin_fry_sketch_nucleus_sces))] <- 
+  alevin_fry_sketch_nucleus_sces
+```
+
+When we read in the counts data that has been processed with alevin-fry using the --unfiltered flag rather than using --knee-distance, all cells that are identified in the 10X whitelist are output. 
+This means that we are going to have cells that are not considered 'true' cells and we need to filter them out before doing any comparisons across the tools. 
+We could use `DropletUtils::barcodeRanks` but even after using that we still see some samples with 50,000 cells above the knee. 
+According to the documenation for DropletUtils another recommendation is to use their method for detecting [empty droplets](https://www.bioconductor.org/packages/devel/bioc/vignettes/DropletUtils/inst/doc/DropletUtils.html#detecting-empty-droplets). 
+Here, I am implementing `DropletUtils::emptyDrops` for the alevin-fry samples used with the `--unfiltered` flag. 
```

Comment:
For a fair comparison, we should probably do this for the kallisto samples as well.